### PR TITLE
feat(console-lite): add trade balance component and hook

### LINKED ISSUE #343: View my balance I hold for the settlement asset
## Task
As a user I would like to see my balance I hold for the settlement asset

## ACs
The balance should be shown for the settlement asset

### PR DESCRIPTION
**Pull request by elmar-vega**
# Related issues 🔗

Closes #343 

# Description ℹ️

Displays balance of settlement asset for the selected key and trade

# Demo 📺
Screenshot 2022-06-07 at 16 42 09

# Technical 👨‍🔧

- Added useSettlementBalance Hook and Spec
- Added new fields to query for DealTicketQuery
- Created local DealTicketBalance component
- Regenerated types
- Added PartyBalanceQuery to local DealTicketContainer

### CODE REVIEW

**Review by notbot00 — APPROVED** [review]

File: apps/simple-trading-app/src/app/components/deal-ticket/deal-ticket-container.tsx [hunk 1]
```diff
@@ -4,15 +4,52 @@ import {
...
+      }
+    }
+  }
+`;
+
 export const DealTicketContainer = () => {
   const { marketId } = useParams<{ marketId: string }>();
+  const { keypair } = useVegaWallet();
+
+  const { data: partyData, loading } = useQuery(PARTY_BALANCE_QUERY, {
```
Comment (mattrussell36) [inline apps/simple-trading-app/src/app/components/deal-ticket/deal-ticket-container.tsx:34]: suggestion (non blocking): you should be able to run nx run types:generate to generate types for this query which you can pass in as type arguments to useQuery

File: apps/simple-trading-app/src/app/hooks/use-settlement-account.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import type { PartyBalanceQuery_party_accounts } from '../components/deal-ticket/__generated__/PartyBalanceQuery';
+import { render, screen } from '@testing-library/react';
+
+interface Props {
+  settlementAssetId: string;
+  accounts: PartyBalanceQuery_party_accounts[];
+}
+
+const MockComponent = ({ settlementAssetId, accounts }: Props) => {
+  const settlementAccount = useSettlementAccount(settlementAssetId, accounts);
```
Comment (mattrussell36) [inline apps/simple-trading-app/src/app/hooks/use-settlement-account.spec.tsx:12]: suggestion (non blocking): We are already using `@testing-library/react-hooks` if you want to use it!

**Review by mattrussell36 — APPROVED** [review]

File: apps/simple-trading-app/src/app/hooks/use-settlement-account.spec.tsx (hunk 2 above)
Comment (elmar-vega) [reply apps/simple-trading-app/src/app/hooks/use-settlement-account.spec.tsx:12]: Nice! Will refactor.

File: apps/simple-trading-app/src/app/components/deal-ticket/deal-ticket-container.tsx (hunk 1 above)
Comment (elmar-vega) [reply apps/simple-trading-app/src/app/components/deal-ticket/deal-ticket-container.tsx:34]: I did run it, might have forgot to add it. Will refactor :)

File: apps/simple-trading-app/src/app/hooks/use-settlement-account.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import type { PartyBalanceQuery_party_accounts } from '../components/deal-ticket/__generated__/PartyBalanceQuery';
+import { useMemo } from 'react';
+
+export const useSettlementAccount = (
+  settlementAssetId: string,
+  accounts: PartyBalanceQuery_party_accounts[] | null
+): PartyBalanceQuery_party_accounts | null => {
+  const callback = () =>
+    accounts?.length
```
[the remark this answers is not in the document]
Comment (elmar-vega) [reply apps/simple-trading-app/src/app/hooks/use-settlement-account.ts]: Done

**Pull request merged**